### PR TITLE
feat(transfer): Add transfer call on treasury

### PR DESCRIPTION
**Pull request by filip88**

### CODE REVIEW

File: src/apis/treasury.ts [hunk 1]
```diff
@@ -24,4 +37,37 @@ export class DefaultTreasuryAPI implements TreasuryAPI {
...
+        }
+
+        return false;
+    }
+
+    /**
+     * @param destination The address of a user
+     * @param amount The Amount for a transfer
+     */
+    async transferPolkaBTC(destination: string, amount: string): Promise<void> {
```
Comment (nud3l) [inline src/apis/treasury.ts:61]: can you rename to "transfer" please?

File: src/apis/treasury.ts [hunk 2]
```diff
@@ -24,4 +37,37 @@ export class DefaultTreasuryAPI implements TreasuryAPI {
...
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param destination The address of a user
+     * @param amount The Amount for a transfer
```
Comment (nud3l) [inline src/apis/treasury.ts:59]: please rephrase: `@param amount - the amount in satoshi to transfer`

Comment (savudani8) [reply src/apis/treasury.ts:59]: Also renaming the parameter to `amountSatoshi` would be nice

File: src/apis/treasury.ts [hunk 3]
```diff
@@ -1,13 +1,26 @@
 import { AccountId, Balance } from "@polkadot/types/interfaces/runtime";
 import { ApiPromise } from "@polkadot/api";
+import { sendLoggedTx } from "../utils";
+import { AddressOrPair } from "@polkadot/api/submittable/types";
+import { EventRecord } from "@polkadot/types/interfaces";
 
 export interface TreasuryAPI {
     totalPolkaBTC(): Promise<Balance>;
     balancePolkaBTC(id: AccountId): Promise<Balance>;
+    transferPolkaBTC(destination: string, amount: String): Promise<void>;
```
Comment (savudani8) [inline src/apis/treasury.ts]: I re-checked the type of the `api.tx.polkaBtc.transfer` and the second parameter (the `amount`) is of type `Compact<Balance> | AnyNumber | Uint8Array`, where `AnyNumber = BN | BigInt | Uint8Array | number | string;`. So the call should work with the lowercase `string` type

File: src/apis/treasury.ts [hunk 4]
```diff
@@ -24,4 +37,37 @@ export class DefaultTreasuryAPI implements TreasuryAPI {
...
+        }
+
+        return false;
+    }
+
+    /**
+     * @param destination The address of a user
+     * @param amount The amount in satoshi to transfer
+     */
+    async transfer(destination: string, amount: string): Promise<void> {
```
Comment (nud3l) [inline src/apis/treasury.ts:61]: as Daniel mentioned, would be good to rename this to `amountSatoshi`

**Review by nud3l — APPROVED** [review]

**Comment by savudani8** [comment]
The build is also failing because the mocked functions are empty. You can just use a `return;` statement to shut up the linter

**Review by nud3l — APPROVED** [review]

**Pull request merged**